From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 13 Jan 2017 21:16:30 +0100
Subject: [PATCH] patch-packages: disable for Armband

Armband repackages cirros-testvm to add AArch64 binaries, and hence
rebuilds x86 binaries too, to align package versions.

Account for Armband's cirros_testvm version by using its specific
version string instead of the old MOS one.

Now Armband also patches test-vm with an updated x86 image in the
upstream repo, so we can disable patch-packages for cirros-testvm
during ISO build.

Since cirros-testvm was the only consumer of patch-packages,
disable it completely for Armband.

JIRA: ARMBAND-200

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 build/install/install.sh                    | 10 +++++-----
 build/patch-packages/Makefile               |  3 ++-
 build/patch-packages/cirros_testvm/Makefile |  2 +-
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/build/install/install.sh b/build/install/install.sh
index 48707a9..2d27e26 100755
--- a/build/install/install.sh
+++ b/build/install/install.sh
@@ -408,6 +408,8 @@ copy_packages() {
         printf "\n\n" | tee -a  $REPORTFILE
         for line in `cat $TOP/patch-packages/release/patch-replacements`
         do
+            echo "Did not expect a line here, not supported"
+            exit 1
             echo "Processing $line ..."
             frompkg=`echo $line | cut -d ">" -f 1`
             topkg=`echo $line | cut -d ">" -f 2`
@@ -453,7 +455,7 @@ copy_packages() {
     APT_DEB_CONF="$TOP/install/apt-ftparchive-deb.conf"
     APT_UDEB_CONF="$TOP/install/apt-ftparchive-udeb.conf"

-    apt-ftparchive -c "${APT_REL_CONF}" generate "${APT_DEB_CONF}"
+    echo Not running apt-ftparchive -c "${APT_REL_CONF}" generate "${APT_DEB_CONF}"
     echo Not running apt-ftparchive generate "${APT_UDEB_CONF}"

     # Fuel also needs this index file
@@ -462,10 +464,8 @@ copy_packages() {
     # /^Version:/{print pkg ": \"" $2 "\""}' > ubuntu-versions.yaml
     # cp ubuntu-versions.yaml $DEST

-    echo apt-ftparchive -c "${APT_REL_CONF}" release dists/mos10.0/ > dists/mos10.0/Release
-    apt-ftparchive -c "${APT_REL_CONF}" release dists/mos10.0/ > dists/mos10.0/Release
-    echo gzip -9cf dists/mos10.0/Release > dists/mos10.0/Release.gz
-    gzip -9cf dists/mos10.0/Release > dists/mos10.0/Release.gz
+    # apt-ftparchive -c "${APT_REL_CONF}" release dists/mos10.0/ > dists/mos10.0/Release
+    # gzip -9cf dists/mos10.0/Release > dists/mos10.0/Release.gz

     popd > /dev/null

diff --git a/build/patch-packages/Makefile b/build/patch-packages/Makefile
index aaac038..bb450ad 100644
--- a/build/patch-packages/Makefile
+++ b/build/patch-packages/Makefile
@@ -8,7 +8,8 @@
 # http://www.apache.org/licenses/LICENSE-2.0
 ##############################################################################

-SUBDIRS := cirros_testvm
+# NOTE(armband): Fuel@OPNFV patches cirros_testvm @ ISO build, we don't.
+SUBDIRS :=
 SUBCLEAN = $(addsuffix .clean,$(SUBDIRS))

 .PHONY: $(SUBDIRS) $(SUBCLEAN) clean
diff --git a/build/patch-packages/cirros_testvm/Makefile b/build/patch-packages/cirros_testvm/Makefile
index b6a56d4..b0163d0 100644
--- a/build/patch-packages/cirros_testvm/Makefile
+++ b/build/patch-packages/cirros_testvm/Makefile
@@ -23,7 +23,7 @@ clean:

 .PHONY: release
 release:
-	../tools/deb_unpack c/cirros-testvm/cirros-testvm_0.3.4-2~u16.04+mos5_amd64.deb $(ORIGISO)
+	../tools/deb_unpack c/cirros-testvm/cirros-testvm_0.3.4-2+amos2~u16.04+mos5_amd64.deb $(ORIGISO)
 	@rm -rf package/usr/share/cirros-testvm/*
 	wget -O package/usr/share/cirros-testvm/cirros-x86_64-disk.img http://download.cirros-cloud.net/daily/20160722/cirros-d160722-x86_64-disk.img
 	../tools/deb_pack $(REVSTATE)
